src/main/java/main/movie/security/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package main.movie.security.impl;
+
+import java.util.Optional;
+import main.movie.exceptions.AuthenticationException;
+import main.movie.lib.Inject;
+import main.movie.lib.Service;
+import main.movie.model.User;
+import main.movie.security.AuthenticationService;
+import main.movie.service.UserService;
+import main.movie.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> optionalUser = userService.findByEmail(email);
+        if (optionalUser.isEmpty()) {
+            throw new AuthenticationException("There are no user with such login!");
+        }
+        User user = optionalUser.get();
+        if (user.getPassword().equals(HashUtil.hashPassword(password, user.getSalt()))) {
+            return user;
+        }
+        throw new AuthenticationException("Wrong login for that user!");
+    }
+
+    @Override
+    public User register(String email, String password) {
+        User user = new User();
+        user.setEmail(email);
+        user.setSalt(HashUtil.generateSalt());
+        user.setPassword(HashUtil.hashPassword(password, user.getSalt()));
```

Comment:
we have to do hashing and salting password in userservice

Reply:
> we have to do hashing and salting password in userservice

\+